lib/src/extension.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'default_extension_map.dart';
+
+/// Add an override for common extensions since different extensions may map
+/// to the same MIME type.
+final Map<String, String> _preferredExtensionsMap = <String, String>{
+  'application/vnd.ms-excel': 'xls',
+  'application/vnd.ms-powerpoint': 'ppt',
+  'image/jpeg': 'jpg',
+  'image/tiff': 'tif',
+  'image/svg+xml': 'svg',
+  'text/calendar': 'ics',
+  'text/javascript': 'js',
+  'text/plain': 'txt',
+  'text/sgml': 'sgml',
+  'text/x-pascal': 'pas',
+  'video/mp4': 'mp4',
+  'video/mpeg': 'mpg',
+  'video/quicktime': 'mov',
+  'video/x-matroska': 'mkv',
+};
+
+/// Lookup file extension for a given MIME type.
+///
+/// If there are multiple extensions for [mimeType], return preferred extension
+/// if defined, or the first occurrence in the map.
+/// If no extension is found, `null` is returned.
+String? extensionFromMimeOrNull(String mimeType) {
+  final mimeTypeLC = mimeType.toLowerCase();
+  if (_preferredExtensionsMap.containsKey(mimeTypeLC)) {
+    return _preferredExtensionsMap[mimeTypeLC]!;
+  }
+  return defaultExtensionMap.entries
```

Comment:
I'd just inline that here:
```dart
  for (var entry in defaultExtensionsMap.entries) {
     if (entry.value == mimeTypeLC) return entry.key);  
  }
  return null;
````
It's far too much syntactic overhead to add your own `firstWhereOrNull` extension (takes up nine lines further down) to save, basically, one line of code here. And I personally don't find it easier to read either.

Reply:
done